docs/ldml/tr35-keyboards.md

```diff
@@ -2030,6 +2039,221 @@ Footnotes:
+
+### Test Element: tests
+
+> <small>
+>
+> Parents: [keyboardTest](#test-element-keyboardtest)
+>
+> Children: [_special_](tr35.md#special), [test](#test-element-test)
+>
+> Occurrence: Optional, Multiple
+> </small>
+
+This element specifies a particular suite of `<test>` elements.
+
+_Attribute:_ `name`
+
+This attribute specifies a unique name for this suite of tests. These names could be output to the user during test execution, used to summarize success and failure, or used to select or deselect test components.
+
+### Test Element: test
+
+> <small>
+>
+> Parents: [tests](#test-element-tests)
+>
+> Children: [startContext](#test-element-startContext), [event](#test-element-event), [_special_](tr35.md#special)
+>
+> Occurrence: Required, Multiple
+> </small>
+
+This attribute specifies a specific isolated regression test. Multiple test elements do not interact with each other.
+
+_Attribute:_ `name`
+
+This attribute specifies a unique name for this particular test. These names could be output to the user during test execution, used to summarize success and failure, or used to select or deselect test components.
+
+### Test Element: startContext
+
+This element specifies pre-existing text in a document, as if prior to the user’s insertion point. This is useful for testing transforms and reordering. If not specified, the startContext can be considered to be the empty string ("").
+
+> <small>
+>
+> Parents: [test](#Element_test)
+>
+> Children: _none_
+>
+> Occurrence: Optional, Single
+> </small>
+
+_Attribute:_ `to`
+
+Specifies the starting context. This text may be escaped with `\u` notation, see [Escaping](#Escaping).
+
+### Test Element: event
+
+> <small>
+>
+> Parents: [test](#test-element-test)
+>
+> Children: _none_
+>
+> Occurrence: Required, Multiple
+> </small>
+
+This element represents a single keystroke or other gesture event, as well as the required output when the event has occurred. The order of these elements is significant, with cumulative effects: they must be processed from first to last.
+
+The `result` attribute is required, as well as exactly one of either `to` or `key`. Optionally, exactly one of the gesture attributes `flick`, `longPress`, or `tapCount` may be specified. If none of the gesture attributes are specified, then a regular keypress is effected on the key.
+
+See the examples section below for all valid attribute combinations. In reviewing the attributes, see also [`<key>`](#Element_key).
+
+If a key is not found, or a particular gesture has no definition, the output should be behave as if the user performed such an action.  For example, an unspecified flick would result in no output.
+
+_Attribute:_ `to`
+
+This attribute specifies a key by means of the key’s `to` attribute.
+
+It is an error to use this attribute with a `to` value that matches more than one key.  Such keys are not invalid, however, they must be selected here by using the `key` attribute and specifying their key id.
```

Comment:
Is it necessary for this to be an error? We could just say that `flick`, `longPress` and `tapCount` are not permitted with `to` -- which means we can setup tests which are agnostic about how the `to` is generated.

Reply:
I could see that.